runtime/onert/core/src/ir/train/UseDefGenerator.test.cc

```diff
@@ -0,0 +1,2027 @@
+    tgraph.operands().at(ba_input2).data(std::make_unique<ExternalData>(
+      reinterpret_cast<uint8_t *>(data.data()), data.size() * sizeof(float)));
+
+    tgraph.addInput({input});
+    tgraph.addInput({y_true});
+    tgraph.addOutput({output});
+
+    const auto fc_index =
+      addFullyConnectedOperation(tgraph, {input, weights, OperandIndex{}}, {ba_input1});
+
+    operation::BinaryArithmetic::Param param;
+    param.arithmetic_type = operation::BinaryArithmetic::ArithmeticType::ADD;
+    param.activation = Activation::NONE;
+    const auto ba_op = operation::BinaryArithmetic({ba_input1, ba_input2}, {y_pred}, param);
+    const auto ba_index =
+      tgraph.addOperation(std::make_unique<train::operation::BinaryArithmetic>(ba_op));
+
+    const auto loss_index = addLossOperation(tgraph, {y_pred, y_true}, {output});
+
+    enableAllBackwarding(tgraph);
+
+    EXPECT_NO_THROW(tgraph.setTrainingUseDefs(train::UseDefGenerator{tgraph}()));
+    EXPECT_NO_THROW(tgraph.verify());
+
+    train::TrainingOperationIndex forwarding_fc_index{fc_index, true};
+    train::TrainingOperationIndex forwarding_ba_index{ba_index, true};
+    train::TrainingOperationIndex forwarding_loss_index{loss_index, true};
+    train::TrainingOperationIndex backwarding_fc_index{fc_index, false};
+    train::TrainingOperationIndex backwarding_ba_index{ba_index, false};
+    train::TrainingOperationIndex backwarding_loss_index{loss_index, false};
+
+    std::vector<train::TrainingOperationIndex> expected_forwarding_input_uses{forwarding_fc_index,
+                                                                              backwarding_fc_index};
+    std::vector<train::TrainingOperationIndex> expected_forwarding_input_def{};
+    const auto expected_forwarding_input = createUseDefChain(
+      tgraph.operands().at(input), expected_forwarding_input_uses, expected_forwarding_input_def);
+
+    std::vector<train::TrainingOperationIndex> expected_forwarding_weights_uses{
+      forwarding_fc_index, backwarding_fc_index};
+    std::vector<train::TrainingOperationIndex> expected_forwarding_weights_def{};
+    const auto expected_forwarding_weights =
+      createUseDefChain(tgraph.operands().at(weights), expected_forwarding_weights_uses,
+                        expected_forwarding_weights_def);
+
+    // forwarding output of FC is not used in backwarding if activation is none
+    std::vector<train::TrainingOperationIndex> expected_forwarding_ba_input1_uses{
+      forwarding_ba_index, backwarding_ba_index};
+    std::vector<train::TrainingOperationIndex> expected_forwarding_ba_input1_def{
+      forwarding_fc_index};
+    const auto expected_forwarding_ba_input1 =
+      createUseDefChain(tgraph.operands().at(ba_input1), expected_forwarding_ba_input1_uses,
+                        expected_forwarding_ba_input1_def);
+
+    std::vector<train::TrainingOperationIndex> expected_forwarding_ba_input2_uses{
+      forwarding_ba_index, backwarding_ba_index};
```

Comment:
This expected "uses" means forwarding "add" node and backwarding "add" node are added as "uses" of second input of forwarding "add" node.